### PR TITLE
battery_level.py recommended changes

### PR DESCRIPTION
**Pull request by inclementweather**
Module battery_level.py lines 343 and 350 raise an AttributeError when the Py3status class is instantiated and the battery is below either threshold. 
The self.last_known_status variable isn't declared yet so the threshold conditionals fail. Otherwise if the battery starts above the threshold and drops below it, the error is avoided.
Declaring last_known_status = '' at the top of Py3status fixes the problem.

A subprocess.call() to notify-send on lines 159 and 364 of battery_level.py raises a FileNotFoundError (python3.5) when libnotify is not installed. Neither libnotify nor dunst is listed in the module requirements.
The module should probably use the py3status notify_user() method so people can utilize the -b flag.

### CODE REVIEW

**Comment by inclementweather** [comment]
Ok!  So docstring errors with the way I declared last_known_status at first, I didn't exactly understand the build failure message, but after realizing my error I modified it to use getattr() at the top of the method to check and set a default value.  Just some minor formatting issues after that. I tested everything on my laptop with and without the -p flag, everything seems to be working.

**Comment by tobes** [comment]
@inclementweather thanks for this fix.  Also thanks for the feedback regarding the new docstring tests.  I'll try to get the documentation around them added/improved.

**Comment by inclementweather** [comment]
@tobes My pleasure. This was a good exercise for me. 

Also, I had thought about this more and realized I could declare the last_known_status variable in a class __init__ instead of creating a new variable in the method.  Would that be the more pythonic solution?

**Comment by tobes** [comment]
> Would that be the more pythonic solution? 

Yes it probably would be, but there are always lots of different ways to achieve the same thing :)

You can also update existing commits in git via something like the following
```
git add -p
git commit --amend
git push -f
```

**Comment by inclementweather** [comment]
Ok. I'll make those changes and update this pull request.  Thanks for the tip on amending my commits.

**Comment by inclementweather** [comment]
Modified and tested.  Amended the last CL.

**Comment by tobes** [comment]
This is now a much cleaner looking patch :+1: 

I'm not happy that github has not noticed the code has changed since my last review.  This sort of makes the feature worthless in my opinion.

**Comment by inclementweather** [comment]
I thought that was kind of weird too.

File: py3status/core.py [hunk 1]
```diff
@@ -488,13 +488,16 @@ def log(self, msg, level='info'):
...
             syslog(level, u'{}'.format(msg))
         else:
-            with open(self.config['log_file'], 'a') as f:
+            # Binary mode so fs encoding setting is not an issue
+            with open(self.config['log_file'], 'ab') as f:
                 log_time = time.strftime("%Y-%m-%d %H:%M:%S")
                 out = u'{} {} {}\n'.format(log_time, level.upper(), msg)
                 try:
+                    # Encode unicode strings to bytes
+                    f.write(out.encode('utf-8')
```
Comment (tobes) [inline py3status/core.py]: you've failed to close this, you need to add `)`

**Comment by tobes** [comment]
@inclementweather is it possible for you to revert that last commit in this PR and do it as a separate PR.  It makes it easier to review changes that way and allows some PRs to be merged and others not

You'll need to create a new branch for that PR.

In the future he easiest way is keep you master branch the same as upstream master and then do something like
```
git checkout master  # make sure we are on the master branch
git checkout -b my-new-branch  # create a new branch
# make changes
git add -p # add changes
git commit -m'my commit message'
git push # push your new branch you'll have to set an upstream but git will prompt you how
```
Your master currently contains these extra commits so that won't work for you here.

To fix your master will a pain as really you want to reset it to commit c2fcb866e2 and then make branches from there (or at least branch from there and then add the commits via cherry picking etc).  If you are happy with doing that it'd be great but if it is too complicated (not sure if you've used git much) I can fix it up as I think that both of these solutions are good (excluding the minor bugfix).

It maybe that @ultrabug is happy to take this as is, but in future separate branches is the way to go.

File: py3status/core.py (hunk 1 above)
Comment (inclementweather) [inline py3status/core.py]: Thanks. Looking at how to revert now.

**Comment by ultrabug** [comment]
Thanks a lot @inclementweather for your reports, debugging and cool work on fixing stuff. It is not everyday that I see first contributions being on the py3status core :+1: 

@tobes I'll be okay to merge this as-is for this time, I'm sure @inclementweather will think about separate PRs for his next contributions ! :)

Also friendly thanks @maximbaz for the review.

@tobes this PR looks ready for merge to me, am I wrong ?

**Comment by ultrabug** [comment]
If you participate in hacktoberfest, I've flagged this PR ;)

**Comment by inclementweather** [comment]
Sorry! I didn't realize my newest commit would go to the pull request too.

**Comment by tobes** [comment]
@ultrabug 

> @tobes this PR looks ready for merge to me, am I wrong ?

I think for merging straight after 3.2

@inclementweather 
> Sorry! I didn't realize my newest commit would go to the pull request too. 

It can take a little while to work out how things work, but it gets easy quickly

**Comment by inclementweather** [comment]
@tobes Thanks. I appreciate the git help.  

I'll wait until the changes are merged then reset.

**Comment by tobes** [comment]
> I'll wait until the changes are merged then reset.

Once they are merged then your master will be happy again because it will have the same commits as master here.  To get the latest stuff from master you'll do something like this
```
git checkout master  #make sure we are on master branch
git pull upstream master  # pull latest upstream
```
the odd `git fetch upstream --tags` once in a while is also good to get any upstream tags (version releases are tagged)

**Comment by inclementweather** [comment]
> git checkout master  #make sure we are on master branch
> git pull upstream master  # pull latest upstream

@tobes would it revert my commits if I did that now?

**Comment by tobes** [comment]
> @tobes would it revert my commits if I did that now? 

No it would just get the latest commits in master here.  The only time it would be an issue is if there was a merge conflict but you'll be fine here.  You will find that there will be commits after your ones if you do this `git log` will show the commits in order.

Generally keep your master branch as a copy of upstream and do any changes in fresh branches.

Git is pretty good at sorting things out so as long as there are not changes in the same bit of code you are generally fine.  merge conflicts can be a little confusing to start with but are fine once you've got used to them.

**Comment by ultrabug** [comment]
Thanks a lot @inclementweather et al for this PR and review :+1:

**Pull request merged**